package/kernel/linux/modules/lib.mk

```diff
@@ -98,6 +98,132 @@ endef
 $(eval $(call KernelPackage,lib-crc32c))
 
 
+define KernelPackage/lib-crypto-blake2s
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=BLAKE2s hash function library
+  KCONFIG:=CONFIG_CRYPTO_LIB_BLAKE2S
+  FILES:= \
+	$(LINUX_DIR)/lib/crypto/libblake2s.ko \
+	$(LINUX_DIR)/lib/crypto/libblake2s-generic.ko
+endef
+
+define KernelPackage/lib-crypto-blake2s/x86/64
+  KCONFIG+=CONFIG_CRYPTO_BLAKE2S_X86
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/blake2s-x86_64.ko
+endef
+
+$(eval $(call KernelPackage,lib-crypto-blake2s))
+
+
+define KernelPackage/lib-crypto-chacha20
+  SUBMENU:=$(LIB_MENU)
+  TITLE:=ChaCha library interface
+  KCONFIG:=CONFIG_CRYPTO_LIB_CHACHA
+  FILES:=$(LINUX_DIR)/lib/crypto/libchacha.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/x86/64
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_X86_64
+  FILES+=$(LINUX_DIR)/arch/x86/crypto/chacha-x86_64.ko
+endef
+
+define KernelPackage/lib-crypto-chacha20/arm-neon
+  KCONFIG+=CONFIG_CRYPTO_CHACHA20_NEON
+  FILES+=$(LINUX_DIR)/arch/arm/crypto/chacha-neon.ko
+endef
+
+KernelPackage/lib-crypto-chacha20/imx6=$(KernelPackage/lib-crypto-chacha20/arm-neon)
+KernelPackage/lib-crypto-chacha20/ipq40xx=$(KernelPackage/lib-crypto-chacha20/arm-neon)
+KernelPackage/lib-crypto-chacha20/mvebu/cortexa9=$(KernelPackage/lib-crypto-chacha20/arm-neon)
+
+define KernelPackage/lib-crypto-chacha20/arm64-neon
```

Comment:
While there's no scalar fallback code for !neon systems, I wonder if the "arm64-neon" designation is your own invention and if this should actually be called "arm64" to be simpler, or if "arm64-neon" is a system-wide thing for platforms that only have neon.

Reply:
In fact, I don't see arm64 mentioned in any kernel module makefile, so we get to blaze a trail here. The name doesn't really matter, it's the target mapping below that decides if a particular target will take that definition:
`KernelPackage/lib-crypto-chacha20/mvebu/cortexa9=$(KernelPackage/lib-crypto-chacha20/arm-neon)`

Reply:
here is the full logic from include/kernel.mk:
```
201   define Package/kmod-$(1)
202     TITLE:=$(TITLE)
203     SECTION:=kernel
204     CATEGORY:=Kernel modules
205     DESCRIPTION:=$(DESCRIPTION)
206     EXTRA_DEPENDS:=kernel (=$(LINUX_VERSION)-$(LINUX_RELEASE)-$(LINUX_VERMAGIC))
207     VERSION:=$(LINUX_VERSION)$(if $(PKG_VERSION),+$(PKG_VERSION))-$(if $(PKG_RELEASE),$(PKG_RELEASE),$(LINUX_RELEASE))
208     PKGFLAGS:=$(PKGFLAGS)
209     $(call KernelPackage/$(1))
210     $(call KernelPackage/$(1)/$(BOARD))
211     $(call KernelPackage/$(1)/$(BOARD)/$(if $(SUBTARGET),$(SUBTARGET),generic))
212   endef
```

Reply:
I surmise from your previous question that the `-neon` vs no`-neon` suffix designation by you was intentional and that you want to keep this convention. That's fine with me. Just wanted to check.

Reply:
I changed it to arm64, but like I said above, no targets are currently mapped to this arm64 definition.

Reply:
Alright. Sounds good.